### ISSUE TITLE
Probabilistic Forecasting using NGboost

### ISSUE THREAD

**Issue by satya-pattnaik** [alan-turing-institute/sktime#855]
**Is your feature request related to a problem? Please describe.**
Can we build a Probabilistic Forecaster using Ngboost. A Probabilistic regression like Ngboost method will give us the confidence intervals out of the box unlike others where we need a heuristic(like quantile loss) to get the values of Conf/Pred Ints.

**Describe the solution you'd like**
A rough sketch:
```
class ProbForecaster:
     def __init__():
           self.estimator_ = NgboostRegressor()
     def fit(y,X=None):
           y = make_reduction(y)
           self.estimator_.fit(y)
     def predict(fh,X=none,alpha=0.95,conf_int=True):
           mean_preds =  self.estimator_.predict(y)
           distribution =  self.estimator_.pred_dist(y)
           conf_ints = scipy.norm.interval(alpha, distribution.loc, distribution.scale )
           return mean_preds,conf_ints
```

**Describe alternatives you've considered**
Using Xgboost/LightGBM with quantile loss makes the calculation of prediction intervals inefficient.

**Additional context**
Ngboost Doc- [https://stanfordmlgroup.github.io/projects/ngboost/](https://stanfordmlgroup.github.io/projects/ngboost/)

**Comment by fkiraly** [comment]
Interesting!
NGboost is a probabilistic prediction algorithm, not a forecaster - so we would first have to build a full probabilistic interface. That would be a great thing to have for `sktime`.

Now about something that's partly funny and partly not funny... 
"NGboost" is basically identidal to the probabilistic boosting algorithm I proposed in section 6.4 of my 2018 paper https://arxiv.org/abs/1801.00753
The probabilistic prediction interface of the NGboost python package closely follows principles in the `sktime` companion package `skpro` (scikit-learn like probabilistic prediction), or the R package `mlr3proba`.

I'm reasonably certain that Ng et al know of both, but still don't cite them... that's not very nice.

Anyway, we should develop `skpro` into the probabilistic scikit-learn that it was meant to be, but unfortunately it's currently without maintainer. Might that be something you would be interested in?

**Comment by satya-pattnaik** [comment]
Yes, 6.4 of the above mentioned paper deals with a similar concept. 
That sounds good to me, I can use Skpro to build it. The question is how to build that probabilistic interface, if I can get some starting point I can carry it from there, as I was planning to use ```make_reduction```(Which I assume now is not the correct way???)

**Comment by fkiraly** [comment]
well, if you want to use `skpro` for probabilistic predictions, the package itself needs an update - it's a bit of a larger project than just adding a method. If you're interested to do that, we should set up a call - also (I believe you are part of the mentoring programme?) discuss with your mentor.

**Comment by satya-pattnaik** [comment]
Yes, will discuss about this with my mentor. Thanks

**Comment by drackham** [comment]
I'm very interested in the capability discussed here. It doesn't appear like there has been any progress here. Would you correct me if that is in fact not the case? Thank you!

**Comment by fkiraly** [comment]
actually yes!

We've been re-working the probabilistic forecasting interface:
https://github.com/sktime/sktime/issues/4359

This will enable using probabilistic supervised learners in compositors like `make_reduction` much more easily.

Want to help work on this, @drackham ? It's a bit of an engineering project, but there's a step-by-step roadmap.

Would be much appreciated!

We'll probably move this a bit over the Easter holdays where the volunteer contributors tend to have more time.

**Comment by fkiraly** [comment]
speaking of which, @frthjf, are you still around?

I would like to move the probabilistic interface into `skpro` within the next year or so and use it as an import.

**Comment by fkiraly** [comment]
> Just to clarify, do I understand correctly that the plan is to resurrect the skpro package which in turn becomes an optional dependency of sktime? I could certainly help with updating the skpro code.

Yes!

For now, I've been working in the `src/proba` module. Have a look and let me now what you think!

The design is a mix of `pandas`, `sklearn` (base interface), and `tensorflow-probability` (parameter broadcast).

I'd like to move it out to `skpro`, and make `skpro` a *core* (not optional) dependency of `sktime`. Both `sktime` and `skpro` would eventually depend on `skbase` which has the base class framework.

**Comment by fkiraly** [comment]
For comments, the topical issue is here, @frthjf:
https://github.com/sktime/sktime/issues/4359

(this issue is about a specific probabilistic forecaster)

**Comment by frthjf** [comment]
I see, in that case, why move this out of `sktime` into an independent `skpro` package if it's required back in anyway? Wouldn't it be easier to port the skpro features into sktime instead? It seems to me that the sktime package already has all the CI and package infrastructure we would have to recreate for a skpro re-release.

**Comment by fkiraly** [comment]
(will continue on #4359 for architecture discussion)
